components/identity-mgt-new/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtConfig.java

```diff
@@ -0,0 +1,151 @@
+import org.wso2.carbon.user.core.jdbc.JDBCRealmConstants;
+import org.wso2.carbon.utils.CarbonUtils;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import java.util.*;
+
+/**
+ * encapsulates recovery config data
+ */
+public class IdentityMgtConfig {
+
+    private static final Log log = LogFactory.getLog(IdentityMgtConfig.class);
+    private static IdentityMgtConfig identityMgtConfig;
+    private boolean accountLockEnable;
+    IdentityMgtService identityMgtService = new IdentityMgtServiceImpl();
+    private boolean enableAuthPolicy;
+    private boolean authPolicyAccountExistCheck;
+    private int authPolicyMaxLoginAttempts;
+    private UserIdentityDataStore identityDataStore;
+    private PolicyRegistry policyRegistry = new PolicyRegistry();
+
+    public IdentityMgtConfig(RealmConfiguration configuration) {
+
+        Properties properties = identityMgtService.addConfigurations(MultitenantConstants.SUPER_TENANT_ID);
+        setConfigurations(configuration, properties);
+
+    }
+
+    public IdentityMgtConfig(RealmConfiguration configuration, int tenantId) {
+
+        Properties properties = identityMgtService.getConfigurations(tenantId);
+        setConfigurations(configuration, properties);
+
+    }
+
+    public void setConfigurations(RealmConfiguration configuration, Properties properties) {
+
+        if (!properties.isEmpty()) {
+
+            try {
+                String authPolicyAccountExistCheck = properties.
+                        getProperty(IdentityMgtConstants.PropertyConfig.AUTH_POLICY_ACCOUNT_EXIST);
+                if (authPolicyAccountExistCheck != null) {
+                    this.authPolicyAccountExistCheck = Boolean.parseBoolean(authPolicyAccountExistCheck.trim());
+                }
+
+                String enableAuthPolicy = properties.
+                        getProperty(IdentityMgtConstants.PropertyConfig.AUTH_POLICY_ENABLE);
+                if (enableAuthPolicy != null) {
+                    this.enableAuthPolicy = Boolean.parseBoolean(enableAuthPolicy.trim());
+                }
+
+                String accountLockEnable = properties.
+                        getProperty(IdentityMgtConstants.PropertyConfig.ACCOUNT_LOCK_ENABLE);
+                if (accountLockEnable != null) {
+                    this.accountLockEnable = Boolean.parseBoolean(accountLockEnable.trim());
+                }
+
+                String maxLoginAttemptProperty = properties.
+                        getProperty(IdentityMgtConstants.PropertyConfig.AUTH_POLICY_ACCOUNT_LOCKING_FAIL_ATTEMPTS);
+                if (maxLoginAttemptProperty != null) {
+                    this.authPolicyMaxLoginAttempts = Integer.valueOf(maxLoginAttemptProperty.trim());
+                }
+
+                if (this.authPolicyMaxLoginAttempts == 0) {
+                    // default value is set
+                    this.authPolicyMaxLoginAttempts = 10;
+                }
+            } catch (Exception e) {
```

Comment:
If possible catch different types of Exceptions one by one. If it throws Exception from the try body only catch Exception.